File: Latihan 3/C++/Pc.cpp
/* Saya Silmi Aulia Rohmah mengerjakan Latihan 3 DPBO 2022 C1 
dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
kecurangan seperti yang telah dispesifikasikan. Aamiin */

#include <iostream>
#include <string>

using namespace std;

class Pc{

    private:

        Processor p;
        Disk d;
        Ram r;
        int totalPrice;

    public:

        Pc(){ //konstruktor kosong

        }

        //metode getter dan setter

        Processor getP(){
            return this->p;
        }

        void setP(Processor p){
            this->p = p;
        }

        Disk getD(){
            return this->d;
        }

        void setD(Disk d){
            this->d = d;
        }

        Ram getR(){
            return this->r;
        }

        void setR(Ram r){
            this->r = r;
        }

        int getTotalPrice(){
            return this->totalPrice;
        }

        void setTotalPrice(int totalPrice){
            this->totalPrice = totalPrice;
        }

        void PrintPC(){ //untuk menampilkan spesifikasi pc
            
            this->p.PrintProcessor();
            this->d.PrintDisk();
            this->r.PrintRam();
            cout << "=====================" << endl;
            cout << "Total Price : " << getTotalPrice() << endl;
        }

        ~Pc(){ //destruktor
            
        }

};

File: Latihan 3/Python/PC.py

# Saya Silmi Aulia Rohmah mengerjakan Latihan 3 DPBO 2022 C1 
# dalam mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan 
# kecurangan seperti yang telah dispesifikasikan. Aamiin 

from Disk import Disk #melakukan import class
from Processor import Processor #melakukan import class
from Ram import Ram #melakukan import class

class PC:


    def __init__(self): #Constructor

        # atribut
        self.p = Processor()
        self.d = Disk()
        self.r = Ram()
        self.totalPrice = 0

    # metode getter dan setter

    def getP(self):
        return self.p
    
    def setP(self, p):
        self.p = p

    def getD(self):
        return self.d
    
    def setD(self, d):
        self.d = d

    def getR(self):
        return self.r
    
    def setR(self, r):
        self.r = r

    def getTotalPrice(self):
        return self.totalPrice
    
    def setTotalPrice(self, totalPrice):
        self.totalPrice = totalPrice


    # untuk print spesifikasi pc
    def PrintPC(self):

        self.p.PrintProcessor()
        self.d.PrintDisk()
        self.r.PrintRam()
        print("=====================")
        print("Total Price : " + str(self.getTotalPrice()))
 